hs_core/hs_rdf.py

```diff
@@ -0,0 +1,358 @@
+
+    def __init__(self, store, max_depth=3):
+        super(HydroPrettyXMLSerializer, self).__init__(store)
+        self.forceRDFAbout = set()
+
+    def serialize(self, stream, base=None, encoding=None, **args):
+        self.__serialized = {}
+        store = self.store
+        self.base = base
+        self.max_depth = args.get("max_depth", 3)
+        assert self.max_depth > 0, "max_depth must be greater than 0"
+
+        self.nm = nm = store.namespace_manager
+        self.writer = writer = XMLWriter(stream, nm, encoding)
+        namespaces = {}
+
+        possible = set(store.predicates()).union(
+            store.objects(None, RDF.type))
+
+        for predicate in possible:
+            prefix, namespace, local = nm.compute_qname(predicate)
+            namespaces[prefix] = namespace
+
+        namespaces["rdf"] = "http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+
+        writer.push(RDF.RDF)
+
+        if "xml_base" in args:
+            writer.attribute(XMLBASE, args["xml_base"])
+
+        writer.namespaces(iter(list(namespaces.items())))
+
+        # Write out subjects that can not be inline
+        for subject in store.subjects():
+            if (None, None, subject) in store:
+                if (subject, None, subject) in store:
+                    self.subject(subject, 1)
+            else:
+                self.subject(subject, 1)
+
+        # write out anything that has not yet been reached
+        # write out BNodes last (to ensure they can be inlined where possible)
+        bnodes = set()
+
+        for subject in store.subjects():
+            if isinstance(subject, BNode):
+                bnodes.add(subject)
+                continue
+            self.subject(subject, 1)
+
+        # now serialize only those BNodes that have not been serialized yet
+        for bnode in bnodes:
+            if bnode not in self.__serialized:
+                self.subject(subject, 1)
+
+        writer.pop(RDF.RDF)
+        stream.write(b"\n")
+
+        # Set to None so that the memory can get garbage collected.
+        self.__serialized = None
+
+    def subject(self, subject, depth=1):
+        store = self.store
+        writer = self.writer
+
+        if subject in self.forceRDFAbout:
+            writer.push(RDF.Description)
+            writer.attribute(RDF.about, self.relativize(subject))
+            writer.pop(RDF.Description)
+            self.forceRDFAbout.remove(subject)
+
+        elif subject not in self.__serialized:
+            self.__serialized[subject] = 1
+            type = first(store.objects(subject, RDF.type))
+
+            try:
+                self.nm.qname(type)
+            except:
+                type = None
+
+            element = type or RDF.Description
+            writer.push(element)
+
+            if isinstance(subject, BNode):
+                def subj_as_obj_more_than(ceil):
+                    return True
+                    # more_than(store.triples((None, None, subject)), ceil)
```

Comment:
This ```def subj_as_obj_more_than(ceil)``` does not make sense to me, and it does not look it is called anywhere. Also, the line 272 can be removed.